python-sdk/src/astro/sql/table.py

```diff
@@ -125,6 +122,42 @@ def name(self, value: str) -> None:
             self._name = value
             self.temp = False
 
+
+@define(slots=False)
+class TempTable(BaseTable):
+    """
+    Internal class to represent a Temporary table
+
+    :sphinx-autoapi-skip:
+    """
+
+    temp: bool = field(default=True)
+
+
+@define(slots=False)
+class Table(BaseTable, Dataset):
+    """
+    User-facing class that has information necessary to access a SQL Table. It is agnostic to the database type.
+    If no name is given, it auto-generates a name for the Table and considers it temporary.
+
+    Temporary tables are prefixed with the prefix TEMP_PREFIX.
+
+    :param name: The name of the database table. If name not provided then it would create a temporary name
+    :param conn_id: The Airflow connection id. This will be used to identify the right database type at the runtime
+    :param metadata: A metadata object which will have database or schema name
+    :param columns: columns which define the database table schema.
+    """
+
+    uri: str = field(init=False)
+    extra: dict | None = field(init=False, factory=dict)
+
+    def __new__(cls, *args, **kwargs):
+        name = kwargs.get("name", "") or (args[0] if len(args) > 0 else "")
+        temp = kwargs.get("temp", False)
+        if temp or (not name or name.startswith("_tmp")):
```

Comment:
```suggestion
        if temp or not name or name.startswith("_tmp"):
```

Reply:
I like the one with brackets better for readability

Reply:
Valid point! :)